src/main/java/address/model/BackingAddressBook.java

```diff
@@ -0,0 +1,74 @@
+package address.model;
+
+import address.model.datatypes.ExtractableObservables;
+import address.model.datatypes.Person;
+import address.model.datatypes.ReadablePerson;
+import address.model.datatypes.Tag;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import java.util.Optional;
+
+/**
+ *
+ */
+class BackingAddressBook implements Model {
```

Comment:
The `Model` and `VisibleModel` interfaces actually only involve retrieval of the internal observablelists. 

I could use those names for the implementing classes instead and rename the interfaces to `ObservableModel` and `ObservableVisibleModel` respectively